controllers/disruption_controller.go

```diff
@@ -1046,10 +1050,34 @@ func (r *DisruptionReconciler) recordEventOnTarget(instance *chaosv1beta1.Disrup
 }
 
 // SetupWithManager setups the current reconciler with the given manager
-func (r *DisruptionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *DisruptionReconciler) SetupWithManager(mgr ctrl.Manager, kubeInformerFactory kubeinformers.SharedInformerFactory) error {
+	podToDisruption := func(c client.Object) []reconcile.Request {
+		// podtoDisruption is a function that maps pods to disruptions. it is meant to be used as an event handler on a pod informer
+		// this function should safely return an empty list of requests to reconcile if the object we receive is not actually a chaos pod
+		// which we determine by checking the object labels for the name and namespace labels that we add to all injector pods
+		disruption := []reconcile.Request{}
+
+		if r.log != nil {
+			r.log.Infow("watching event from pod", "name", c.GetName(), "namespace", c.GetNamespace())
+		}
+
+		labels := c.GetLabels()
+		name := labels[chaostypes.DisruptionNameLabel]
+		namespace := labels[chaostypes.DisruptionNamespaceLabel]
+
+		if name != "" && namespace != "" {
+			disruption = append(disruption, reconcile.Request{NamespacedName: types.NamespacedName{Name: name, Namespace: namespace}})
+		}
+
+		return disruption
+	}
+
+	informer := kubeInformerFactory.Core().V1().Pods().Informer()
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&chaosv1beta1.Disruption{}).
 		Owns(&corev1.Pod{}).
```

Comment:
Is it useful to keep this `Owns` call now that we have the informer?

Reply:
I'll test w/o it to confirm